app/assets/javascripts/discourse/tests/integration/components/header-contents-test.gjs

```diff
@@ -8,24 +8,88 @@ import { setupRenderingTest } from "discourse/tests/helpers/component-test";
 module("Integration | Component | Header | Contents", function (hooks) {
   setupRenderingTest(hooks);
 
-  test("showHeaderSearch", async function (assert) {
-    const site = getOwner(this).lookup("service:site");
-    const toggleNavigationMenu = () => {};
-
-    sinon.stub(site, "mobileView").value(true);
-
-    await render(
-      <template>
-        <Contents
-          @sidebarEnabled={{true}}
-          @toggleNavigationMenu={{toggleNavigationMenu}}
-          @showSidebar={{true}}
-        >test</Contents>
-      </template>
-    );
-
-    assert
-      .dom(".floating-search-input-wrapper")
-      .doesNotExist("it does not display when the site is in mobile view");
+  hooks.beforeEach(function () {
+    this.router = getOwner(this).lookup("service:router");
+    const search = getOwner(this).lookup("service:search");
+    sinon.stub(search, "searchExperience").value("search_field");
+  });
+
+  hooks.afterEach(function () {
+    sinon.restore(); // clean up all stubs
+  });
+
+  module("header search", function () {
+    test("is hidden in mobile view", async function (assert) {
+      const site = getOwner(this).lookup("service:site");
+      sinon.stub(site, "mobileView").value(true);
+
+      await render(<template><Contents /></template>);
+
+      assert
+        .dom(".floating-search-input-wrapper")
+        .doesNotExist("it does not display when the site is in mobile view");
+    });
+
+    module("routes handling", function () {
+      test('is hidden in route "signup"', async function (assert) {
+        sinon.stub(this.router, "currentRouteName").value("signup");
+
+        await render(<template><Contents /></template>);
+
+        assert
+          .dom(".floating-search-input-wrapper")
+          .doesNotExist('it does not display on route: "signup"');
```

Comment:
two minor notes:
* generally, to save use some precious characters 😉, it's good to skip the "it " prefix in assertion descriptions/test messages
* in this change specifically, it'd be fine to skip assertion notes since they pretty much repeat the test name they're in